model/commitq/merge_function_registry.go

```diff
@@ -0,0 +1,31 @@
+package commitq
+
+import (
+	"github.com/evergreen-ci/evergreen/thirdparty"
+)
+
+type MergeFunction interface{}
+
+var mergeMethods map[string]MergeFunction
+
+func init() {
+	mergeMethods = map[string]MergeFunction{
+		"merge": func(title string, message string, sha string) error {
+			return thirdparty.GithubPRMerge(title, message, sha, "merge")
+		},
+		"squash": func(title string, message string, sha string) error {
+			return thirdparty.GithubPRMerge(title, message, sha, "squash")
+		},
+		"rebase": func(title string, message string, sha string) error {
+			return thirdparty.GithubPRMerge(title, message, sha, "rebase")
+		},
+	}
+}
+
+func GetMergeAction(actionString string) MergeFunction {
+	action, ok := mergeMethods[actionString]
+	if !ok {
+		return nil
```

Comment:
I think this should return `(MergeFunction, error)`, since users will be more likely to check an error, but may not check that `MergeFunction` is not nil, which can lead to nil pointer bugs. Checking an error that is returned as the last value of a function is more idiomatic Go.

Reply:
Done.